### ISSUE TITLE
[StaticDateRangePicker] renderInput is required while it's not being used

### ISSUE THREAD

**Issue by ifndefdeadmau5** [mui-org/material-ui-pickers#1931]
![A GIF or MEME to give some spice of the internet](url)

## Environment

| Tech                 | Version |
| -------------------- | ------- |
| @material-ui/pickers | v4.0.0-alpha.8        |
| material-ui          | v4.10.2        |
| TypeScript           |  v3.8.3       |
| React                |   16.9.0      |
| Browser              |    Chrome     |
| Peer library         |"@date-io/date-fns": "^2.6.1"|  |

## Steps to reproduce

1. Provide `renderInput` prop to `<StaticDateRangePicker />` because otherwise it's an error.
2. Input isn't visible anywhere(because it's 'static')

## Expected behavior

`renderInput` shouldn't be a required prop

## Actual behavior

`renderInput` is required

**Comment by oliviertassinari** [comment]
@dmtrKovalenko I find it hard to follow the logic around which prop is spread where, as well as how props are forwarded. What do you think about simplifying the types and spreading logic?

**Comment by dmtrKovalenko** [comment]
It is used. Because `StaticDateRangePicker` can render mobile keyboard input view if **dynamic** prop `displayStaticWrapperAs="desktop"` do not passed. Or even with this prop, but suing `showToolbar`. So it cannot be removed. 

![image](https://user-images.githubusercontent.com/16926049/86253626-74bd5680-bbbd-11ea-8475-a8c254337c76.png)

@oliviertassinari typescript inference logic is really a little bit tricky, but the pickers and wrappers logic is also not really simple. Our design allows to maximally share code and inference all required code.
